yb-voyager/cmd/migration_complexity.go

```diff
@@ -200,3 +210,127 @@ func getComplexityForLevel(level string, count int) string {
+</p>
+`
+
+const explainTemplateText = `Reasoning: {{ .ComplexityRationale }}`
+
+type MigrationComplexityExplanationData struct {
+	Summaries           []MigrationComplexityCategorySummary
+	Complexity          string
+	ComplexityRationale string // short reasoning or explanation text
+}
+
+type MigrationComplexityCategorySummary struct {
+	Category        string
+	TotalIssueCount int
+	ImpactCounts    map[string]int // e.g. {"Level-1": 3, "Level-2": 5, "Level-3": 2}
+}
+
+func buildMigrationComplexityExplanation(sourceDBType string, assessmentReport AssessmentReport, reportFormat string) (string, error) {
+	if sourceDBType != POSTGRESQL {
+		return "", nil
+	}
+
+	var explanation MigrationComplexityExplanationData
+	explanation.Complexity = assessmentReport.MigrationComplexity
+	explanation.ComplexityRationale = migrationComplexityRationale
+
+	explanation.Summaries = buildCategorySummary(assessmentReport.Issues)
+
+	var tmpl *template.Template
+	var err error
+	if reportFormat == "html" {
+		tmpl, err = template.New("Explain").Parse(explainTemplateHTML)
+	} else {
+		tmpl, err = template.New("Explain").Parse(explainTemplateText)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("failed creating the explanation template: %w", err)
+	}
+
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, explanation); err != nil {
+		return "", fmt.Errorf("failed executing the template with data: %w", err)
+	}
+	return buf.String(), nil
+}
+
+func buildRationale(finalComplexity string, l1Count int, l2Count int, l3Count int) string {
+	switch finalComplexity {
+	case constants.MIGRATION_COMPLEXITY_HIGH:
+		return fmt.Sprintf("Found %d Level-2 issue(s) and %d Level-3 issue(s), resulting in HIGH migration complexity", l2Count, l3Count)
+	case constants.MIGRATION_COMPLEXITY_MEDIUM:
+		return fmt.Sprintf("Found %d Level-1 issue(s), %d Level-2 issue(s) and %d Level-3 issue(s), resulting in MEDIUM migration complexity", l1Count, l2Count, l3Count)
+	case constants.MIGRATION_COMPLEXITY_LOW:
+		return fmt.Sprintf("Found %d Level-1 issue(s) and %d Level-2 issue(s), resulting in LOW migration complexity", l1Count, l2Count)
+	}
+	return ""
+}
+
+func buildCategorySummary(issues []AssessmentIssue) []MigrationComplexityCategorySummary {
+	summaryMap := make(map[string]*MigrationComplexityCategorySummary)
+	for _, issue := range issues {
+		if issue.Category == "" {
+			continue // skipping unknown category issues
+		}
+
+		if _, ok := summaryMap[issue.Category]; !ok {
+			summaryMap[issue.Category] = &MigrationComplexityCategorySummary{
+				Category:        issue.Category,
+				TotalIssueCount: 0,
+				ImpactCounts:    make(map[string]int),
+			}
+		}
+
+		summaryMap[issue.Category].TotalIssueCount++
+		summaryMap[issue.Category].ImpactCounts[issue.Impact]++
+	}
+
+	var result []MigrationComplexityCategorySummary
+	for _, summary := range summaryMap {
+		summary.Category = utils.SnakeCaseToTitleCase(summary.Category)
```

Comment:
Instead of using this function we can get the Feature heading for that Enum to be consistent with the heading in report. 
but given that we are anyways changing the UI, we can revisit it later so okay for now.

Reply:
yeah lets revisit this. The category here is being fetched from the analyze schema issue.

I see the same issues of formatting in schema reports. We need to fix the root.